### PR TITLE
feat: BitFlip noise channel

### PR DESCRIPTION
**Pull request by MoHermes**

### CODE REVIEW

File: mpqp/noise/noise_model.py [hunk 1]
```diff
@@ -277,25 +277,103 @@ def to_other_language(
...
-        # generate Kraus operators for bit flip noise
-        # kraus_operators = [
-        #     np.sqrt(1 - self.proba) * np.array([[1, 0], [0, 1]]),  # Identity
-        #     np.sqrt(self.proba) * np.array([[0, 1], [1, 0]])      # Bit flip
-        # ]
-        # return KrausRepresentation(kraus_operators)
-        ...
+    Examples:
+        >>> circuit = QCircuit([H(i) for i in range(3)])
+        >>> bf1 = BitFlip(0.1, [0])
```
Comment (hJaffaliColibritd) [inline mpqp/noise/noise_model.py:297]: Bitflip was not added to the all.py and __init__.py of mpqp.noise ?

File: mpqp/noise/noise_model.py [hunk 2]
```diff
@@ -277,25 +277,103 @@ def to_other_language(
...
+        gates: Optional[list[type[Gate]]] = None,
+    ):
+
+        if not (0 <= prob <= 0.5):
+            raise ValueError(
+                f"Invalid probability: {prob} but should be between 0 and 0.5"
+            )
+
+        if gates is not None:
+            if any(gate.nb_qubits != 1 for gate in gates):
```
Comment (hJaffaliColibritd) [inline mpqp/noise/noise_model.py:329]: If we choose to extend bit-flip to multi qubit gates, then we should remove this

File: mpqp/noise/noise_model.py [hunk 3]
```diff
@@ -277,25 +277,103 @@ def to_other_language(
...
+    Args:
+        prob: Bit flip error probability or error rate (must be within [0, 0.5]).
+        targets: List of qubit indices affected by this noise.
+        gates: List of :class:`Gates<mpqp.core.instruction.gates.gate.Gate>`
+            affected by this noise. Only single-qubit gates are valid.
 
-    #     super().__init__(proba, targets, dimension, gates)
+    Raises:
+        ValueError: When the probability is outside of the expected interval [0, 0.5].
+        ValueError: When no target qubits are specified.
```
[the remark this answers is not in the document]
Comment (hJaffaliColibritd) [reply mpqp/noise/noise_model.py:292]: image

**Review by hJaffaliColibritd — CHANGES_REQUESTED** [review]

**Comment by hJaffaliColibritd** [comment]
Doc and examples and notebook were not updated

**Comment by hJaffaliColibritd** [comment]
Maybe add some simple tests, in mpq/tests, like we did for Depolarizing, just to check that the code is running with BitFlip

**Comment by hJaffaliColibritd** [comment]
Merge with the current version of dev

File: mpqp/noise/noise_model.py (hunk 1 above)
Comment (MoHermes) [reply mpqp/noise/noise_model.py:310]: Added.

File: mpqp/noise/noise_model.py (hunk 2 above)
Comment (MoHermes) [reply mpqp/noise/noise_model.py]: Exactly, got removed

File: mpqp/noise/noise_model.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (MoHermes) [reply mpqp/noise/noise_model.py:305]: The version I was working on it when started the task was without this modification.

File: mpqp/noise/noise_model.py [hunk 4]
```diff
@@ -277,25 +277,103 @@ def to_other_language(
...
-    #     self,
-    #     proba: Union[float, Expr],
-    #     targets: List[int],
-    #     dimension: int = 1,
-    #     gates: List[Gate] = None):
+    Args:
+        prob: Bit flip error probability or error rate (must be within [0, 0.5]).
+        targets: List of qubit indices affected by this noise.
+        gates: List of :class:`Gates<mpqp.core.instruction.gates.gate.Gate>`
+            affected by this noise. Only single-qubit gates are valid.
```
[the remark this answers is not in the document]
Comment (MoHermes) [reply mpqp/noise/noise_model.py:301]: Already started the the implementation of it, modifying `apply_noise_to_braket_circuit`

File: mpqp/noise/noise_model.py (hunk 1 above)
Comment (hJaffaliColibritd) [reply mpqp/noise/noise_model.py:318]: Apparently not @MoHermes

**Comment by hJaffaliColibritd** [comment]
image
Pretty print is not supported @MoHermes

**Pull request merged**